plugins/destination_testing.go

```diff
@@ -3,243 +3,241 @@ package plugins
-		receivedResource = r
-		totalResources++
+	resourcesRead, err := p.readAll(ctx, tables[0], sourceName)
+	if err != nil {
+		return fmt.Errorf("failed to read all: %w", err)
 	}
-	if readErr != nil {
-		return readErr
+
+	if len(resourcesRead) != 2 {
+		return fmt.Errorf("expected 2 resource, got %d", len(resourcesRead))
 	}
-	if totalResources != 1 {
-		return fmt.Errorf("expected 1 resource, got %d", totalResources)
+
+	if resource.Data.Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	}
+
+	if resource2.Data.Equal(resourcesRead[1]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[1])
+	}
+
+	secondSyncTime := syncTime.Add(time.Second).UTC()
+	// write second time
+	if err := p.writeOne(ctx, tables, sourceName, secondSyncTime, resource); err != nil {
+		return fmt.Errorf("failed to write one second time: %w", err)
+	}
+
+	resourcesRead, err = p.readAll(ctx, tables[0], sourceName)
+	if err != nil {
+		return fmt.Errorf("failed to read all second time: %w", err)
+	}
+
+	if len(resourcesRead) != 2 {
+		return fmt.Errorf("expected 2 resources, got %d", len(resourcesRead))
+	}
+
+	if resource.Data.Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	}
+
+	if resource2.Data.Equal(resourcesRead[1]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[1])
+	}
+
+	if !s.skipTestOverWriteDeleteStale {
+		if err := p.DeleteStale(ctx, tables, sourceName, secondSyncTime); err != nil {
+			return fmt.Errorf("failed to delete stale data second time: %w", err)
+		}
 	}
 
-	if receivedResource.TableName != expectedResource.TableName {
-		return fmt.Errorf("expected table name %s, got %s", expectedResource.TableName, receivedResource.TableName)
+	resourcesRead, err = p.readAll(ctx, tables[0], sourceName)
+	if err != nil {
+		return fmt.Errorf("failed to read all second time: %w", err)
 	}
+	if len(resourcesRead) != 1 {
+		return fmt.Errorf("expected 1 resource, got %d", len(resourcesRead))
+	}
+
+	if resource2.Data.Equal(resourcesRead[0]) {
+		return fmt.Errorf("expected data to be %v, got %v", resource.Data, resourcesRead[0])
+	}
+
+	return nil
+}
+
+func (*DestinationPluginTestSuite) destinationPluginTestWriteAppend(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+	spec.WriteMode = specs.WriteModeAppend
+	if err := p.Init(ctx, logger, spec); err != nil {
+		return fmt.Errorf("failed to init plugin: %w", err)
+	}
+	tableName := "cq_test_write_append"
+	table := testdata.TestTable(tableName)
+	syncTime := time.Now().UTC()
+	tables := []*schema.Table{
+		table,
+	}
+	if err := p.Migrate(ctx, tables); err != nil {
+		return fmt.Errorf("failed to migrate tables: %w", err)
+	}
+
+	sourceName := uuid.NewString()
+	resource := schema.DestinationResource{
+		TableName: table.Name,
+		Data:      testdata.TestData(),
```

Comment:
I'm not that happy about different tests sharing the same data. I think it's better to have it duplicated than creating a dependency between the different tests, WDYT?

Reply:
They dont share it. it creates new data every call to `TestData`

Reply:
By share I mean if someone changes `TestData` it impacts all tests (i.e. I can add data to test something and it will break other tests accidentally)

Reply:
Oh you mean the function itself. yes but no one else is using it so let's keep it as is now given we didn't have tests before.